tests: net: socket: socketpair: move to new ztest API

hunterhu3000 opened the pull request:
Move socketpair tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

rlubos reviewed (commented):
Each socket pair test seems to be self-contained, do they need to be executed in certain order?

hunterhu3000 commented:
> Each socket pair test seems to be self-contained, do they need to be executed in certain order?

Yes, the numbers in the test case name can be removed now. And CONFIG_ZTEST_SHUFFLE=y can be set to test this test suite.
Thanks to review.

hunterhu3000 committed:
tests: net: socket: socketpair: move to new ztest API

Move socketpair tests to use new ztest API
TEST=twister -T tests/net/socket/socketpair

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>
